docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +163,157 @@ Note that:
   for instance `tedge/configuration_change//etc/tedge/mosquitto-conf/c8y-bridge.conf`.
 * Since the type of configuration file is used as an MQTT topic name, the characters `#` and `+` cannot be used in a type name.
   If such a character is used in a type name (or in the path of a configuration file without explicit type),
-  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
\ No newline at end of file
+  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` supports two aspects:
+1) **Associating with cloud's child-device twin**<br/>
+Allowing to associate a configuration file with a cloud's child-device twin
+2) **Filetransfer from/to external device**<br/>
+Allowing to consume/provide a configuration file from/to an external device via network
+
+## Details to Aspect 1: Associating with cloud's child-device twin
+
+For aspect (1) the plugin expects for each cloud's child-device twin an individually plugin configuration file (similar to whats described section [Configuration](#configuration) above). Thereby each child-device's configuration file is expected in a subfolder, where each subfolder name is treated as `childid`.
+
+Example:
+
+```
+$ tree /etc/tedge/c8y/
+/etc/tedge/c8y/
+├── c8y-configuration-plugin.toml
+├── child1
+│   └── c8y-configuration-plugin.toml
+└── child2
+    └── c8y-configuration-plugin.toml
+```
+
+Here the plugin serves configuration management for the thin-edge device it-self, for a child-device with childid `child1` and for a child-device with childid `child2`. The contents of all three configuration files follow the details described in section [Configuration](#configuration) above.
+
+## Details to Aspect 2: Filetransfer from/to external device
+
+For aspect (2) there is a proposal, followed by another alternative proposal. Decision has to been taken which proposal to follow.
+
+### Proposal: Filetransfer provided by tedge_agent, based on HTTP
+
+To provide/consume configuration files to/from external devices, the HTTP filetransfer feature of the `tedge_agent` is used.
+  
+The HTTP filetransfer feature of the `tedge_agent` provides the service to transfer files from external devices to the local filesystem of the thin-edge device's, and vice versa.
+
+  * `tedge_agent` serves PUT and GET requests on `http://<thin-edge IP address>/tedge/filetransfer/<path>`
+    * `path` is chosen by the external-device; it may contain folders (e.g. `foo/bar/config-file.conf`)
```

Comment:
I hope you're gonna remove this contract where the paths are chosen by the external device and moving to the temporary file based approach.

Reply:
I introduced now a _temporary file up/downloads_ for the HTTP filetransfer feature. So the external device has no more to deal with paths.